docs/source/cpp/compute.rst

```diff
@@ -234,6 +238,88 @@ Notes:
 
 * \(6) Output is Float64 or input type, depending on QuantileOptions.
 
+* \(7) tdigest/t-digest computes approximate quantiles, and so only needs a
+  fixed amount of memory. See the `reference implementation
+  <https://github.com/tdunning/t-digest>`_ for details.
+
+Grouped Aggregations ("group by")
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Grouped aggregations are not directly invokable, but are used as part of a
+group by operation. Like scalar aggregations, grouped aggregations reduce
+multiple input values to a single output value. Instead of aggregating all
+values of the input, however, grouped aggregations partition of the input
+values on some set of "key" columns, then aggregate each group individually,
+and emit one output per input group.
+
+As an example, for the following table:
+
++-----------------+--------------+
+| Column "x"      | Column "key" |
++=================+==============+
+| 2               | "a"          |
++-----------------+--------------+
+| 5               | "a"          |
++-----------------+--------------+
+| null            | "b"          |
++-----------------+--------------+
+| null            | "b"          |
++-----------------+--------------+
+| null            | null         |
++-----------------+--------------+
+| 5               | null         |
++-----------------+--------------+
+
+We compute a sum of column "x", grouped on the key column "key". This gives us
+three groups:
+
++-----------------+--------------+
+| Column "sum(x)" | Column "key" |
++=================+==============+
+| 7               | "a"          |
++-----------------+--------------+
+| null            | "b"          |
++-----------------+--------------+
+| 5               | null         |
++-----------------+--------------+
+
+The supported aggregation functions are as follows.
+
++---------------+-------+-------------+----------------+----------------------------------+-------+
+| Function name | Arity | Input types | Output type    | Options class                    | Notes |
++===============+=======+=============+================+==================================+=======+
+| hash_all      | Unary | Boolean     | Scalar Int64   | :struct:`ScalarAggregateOptions` | \(1)  |
++---------------+-------+-------------+----------------+----------------------------------+-------+
+| hash_any      | Unary | Any         | Scalar Int64   | :struct:`ScalarAggregateOptions` | \(1)  |
++---------------+-------+-------------+----------------+----------------------------------+-------+
+| hash_count    | Unary | Boolean     | Scalar Int64   | :struct:`CountOptions`           | \(2)  |
++---------------+-------+-------------+----------------+----------------------------------+-------+
+| hash_mean     | Unary | Numeric     | Scalar Float64 |                                  |       |
```

Comment:
Is it accurate that hash_mean and hash_sum don't take ScalarAggregateOptions (yet)?

Reply:
They do, I fixed the table.